pulumi/components/application/apprunner_service.py

```
@@ -0,0 +1,43 @@
+import pulumi
+import pulumi_aws as aws
+
+class AppRunnerServiceComponent(pulumi.ComponentResource):
+    def __init__(self, name: str, ecr_repo_url, apprunner_role_arn, vpc_connector_arn,
+                 environment_variables: dict = None, port: str = "80", tags: dict = None, opts: pulumi.ResourceOptions = None):
```

Ruff (0.15.9)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)